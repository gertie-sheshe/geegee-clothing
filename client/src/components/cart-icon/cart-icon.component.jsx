import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {
  CartContainer,
  ShoppingIcon,
  ItemCountContainer,
} from './cart-icon.styles';

import { toggleCartHidden } from '../../redux/cart/cart.actions';

// Selectors
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

const CartComponent = ({ toggleCartHidden, itemCount }) => (
  <CartContainer onClick={toggleCartHidden}>
    <ShoppingIcon />
    <ItemCountContainer>{itemCount}</ItemCountContainer>
  </CartContainer>
);

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

export default connect(mapStateToProps, mapDispatchToProps)(CartComponent);
