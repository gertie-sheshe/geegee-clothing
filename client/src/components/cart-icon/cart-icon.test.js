import React from 'react';
import { shallow } from 'enzyme';
import { CartIcon } from './cart-icon.component';

describe('CartIcon Component Tests', () => {
  let wrapper;
  let mockToggleCartHidden;

  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden,
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it('Should render CartIcon Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartContainer').simulate('click');
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it('Should render the itemCount as the text', () => {
    const itemCount = parseInt(wrapper.find('ItemCountContainer').text());
    expect(itemCount).toBe(0);
  });
});
