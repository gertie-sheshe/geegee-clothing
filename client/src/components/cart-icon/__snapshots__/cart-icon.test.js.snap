// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component Tests Should render CartIcon Component 1`] = `ShallowWrapper {}`;
